### DISCUSSION TITLE
Blur callback for Autocomplete component

### DISCUSSION THREAD

**Discussion by exrezzo** [Megabit/Blazorise#2495] [Q&A]
Hi all!
I'm using this wonderful component, in my case I extended it by wrapping it in a new component that calls data from database or rest apis, it works quite decently. 
I'm trying to achieve a better control over it: I'd like to implement a callback for when a users blurs the Autocomplete component, in order to prevent the cases where he focuses in (with an already inserted value), writes something but without selecting a new value from the dropdown and then blurs. I would like to check on blur that if the user has not selected a new value then the original one has to be restored. In that way the user is fully aware that he didn't finished to edit the actual value in the Autocomplete.
How can I achieve this? I was trying to extend the Autocomplete component, but honestly I'm not a senior developer and still I don't know how to contribute to this project (I would like to, because it has become my base toolset for developing in my company).

Thank you!

**Comment by stsrki** [comment]
I don't this is possible with the current implementation. If needed we can add a new event-callback OnBlur to Autocomplete. Once we add it, it would be possible to reset change by calling `.Clear()` method.

---

In the meantime maybe `SearchChanged` event can help you? It is raised every time a user writes something in the text field. You can check for valid values in there.

**Reply by exrezzo** [reply]
Thanks for the response. 
That would be a very nice feature, anyways, I don't think that `SearchChanged` che be useful for my purpose, because I needed an event where for example a user starts to edit the value of an Autocomplete but then he leaves it without selecting a new value; in that case, a new "half written" string is showing in the Autocomplete but the underlying selected value it's the same as before, so he is not well aware that he didn't change the value of the Autocomplete.
